spec/facility_spec.rb

```diff
@@ -23,4 +31,79 @@
   end
+
+  describe 'Iteration 2, Vehicle Registration' do
+    it '#register_vehicle' do
+      expect(@facility_1.registered_vehicles).to eq([])
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+      expect(@cruz.registration_date).to eq(Date.today)
+      expect(@cruz.plate_type).to eq(:regular)
+      expect(@facility_1.collected_fees).to eq(100)
+    end
+
+    it 'register multiple vehicles' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+      
+      expect(@facility_1.collected_fees).to eq(325)
+    end
+
+    it 'verify registration service before allowing registration' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+
+      @facility_2.register_vehicle(@camaro)
+      expect(@facility_2.registered_vehicles).to eq([])
+    end
+  end
+
+  describe 'Iteration 2, Drivers License' do
+    it '#administer_written_test if service available' do
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+
+      @facility_1.add_service('Written Test')
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+      expect(@registrant_1.license_data[:written]).to eq(true)
+    end
+
+    it 'must have permit before taking written test' do
+      @facility_1.add_service('Written Test')
+      expect(@registrant_2.permit?).to eq(false)
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(false)
+
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(true)
+      expect(@registrant_2.license_data).to eq({:written=>true, :license=>false, :renewed=>false})
+    end
+  end
+
+  describe 'Iteration 2, Road Test' do
+    it '#administer_road_test only if service available and written test passed' do
+      @facility_1.add_service('Written Test')
+      @facility_1.add_service('Road Test')
+      expect(@facility_1.administer_road_test(@registrant_1)).to eq(false)
+      
+      @facility_1.administer_written_test(@registrant_1)
+      @facility_1.administer_road_test(@registrant_1)
+      expect(@registrant_1.license_data[:license]).to eq(true)
+    end
+  end
+
+  describe 'Iteration 2, Renew License' do #must have service, have license, [and passed road test]<- redundant
+    it '#renew_drivers_license requires service' do
+      expect(@facility_1.renew_drivers_license(@registrant_1)).to eq(false)
+      
+      @facility_1.add_service('Renew License')
+      expect(@facility_1.renew_drivers_license(@registrant_1)).to eq(false)
+      
+      @registrant_1.license_data[:license] = true
+      expect(@facility_1.renew_drivers_license(@registrant_1)).to eq(true)
+    end
+  end
```

Comment:
overall good testing in this class. There are so many possible combinations of logistics going on here and it's important to test them all.